FusionIIIT/applications/academic_information/views.py

```diff
@@ -1,3 +1,391 @@
-# from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.contrib import messages
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import render
+from django.views.decorators.csrf import csrf_exempt
+
+from applications.globals.models import Designation, ExtraInfo
+
+from .forms import MinuteForm
+from .models import (Course, Exam_timetable, Grades, Meeting, Student,
+                     Student_attendance, Timetable)
+
+
+def homepage(request):
+    form = MinuteForm()
+    try:
+        s = Designation.objects.get(name='senate')
+        v = Designation.objects.get(name='Convenor')
+        t = Designation.objects.get(name='Co Convenor')
+        d = Designation.objects.get(name='Dean')
+        senates = ExtraInfo.objects.filter(designation=s)
+        Convenor = ExtraInfo.objects.filter(designation=v)
+        CoConvenor = ExtraInfo.objects.filter(designation=t)
+        Dean = ExtraInfo.objects.get(designation=d)
+        students = Student.objects.filter(id__in=senates)
+        meetings = Meeting.objects.all()
+        student = Student.objects.all()
+        extra = ExtraInfo.objects.all()
+        course = Course.objects.all()
+        timetable = Timetable.objects.all()
+        exam_t = Exam_timetable.objects.all()
+        grade = Grades.objects.all()
+    except:
+        senates = ""
+        students = ""
+        Convenor = ""
+        CoConvenor = ""
+        meetings = ""
+        Dean = ""
+        grade = ""
+        course = ""
+        exam_t = ""
+        timetable = ""
+    pass
```

Comment:
This view function will not move ahead due to this `pass`

Reply:
is there a way to use ajax while pressing the side tabs? Because when i use the tabs that are in {% include xyz.html %} file, it opens for the first time but after that it doesn't.

Reply:
I talked to @anuraagheisenberg and he said, he helped resolve it. I hope it's done.